networkx/utils/mapped_queue.py

```diff
@@ -43,7 +43,13 @@ def __lt__(self, other):
             return self.priority < other
         # assume comparing to another _HeapElement
         if self.priority == other_priority:
-            return self.element < other.element
+            try:
+                return self.element < other.element
+            except TypeError as err:
+                raise TypeError(
+                    f"Consider using a tuple for the priority to avoid such comparisons: "
+                    f"{{k: (v, i) for i, (k,v) in enumerate(data.items())}}"
```

Comment:
Ah sorry, I didn't know that the intent was to actually show the code snippet and not the results - in that case I think it makes much more sense to do away with the fstring altogether!

```suggestion
                    "{k: (v, i) for i, (k,v) in enumerate(data.items())}"
```